### PR TITLE
Fix/evaluation index

### PR DESCRIPTION
**Pull request by florestankorp**

### CODE REVIEW

File: spec/features/evaluation_scoring_spec.rb [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
 
 RSpec.describe Evaluation, type: :model do
-  describe "Navigation teacher evaluation" do
-    let(:user1) { create :user }
-    let!(:student1) { create :student }
-    let!(:teacher1) { create :teacher, user: user1 }
+  let(:user1) { create :user }
+  let!(:student1) { create :student, name: "John Doe" }
+  let!(:teacher1) { create :teacher, user: user1 }
+  let!(:evaluation1) { create :evaluation, score: Evaluation::RED, teacher: teacher1, student:student1 }
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: Space missing after colon.

Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: Line is too long. [104/100]

File: spec/features/evaluation_scoring_spec.rb [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
 
 RSpec.describe Evaluation, type: :model do
-  describe "Navigation teacher evaluation" do
-    let(:user1) { create :user }
-    let!(:student1) { create :student }
-    let!(:teacher1) { create :teacher, user: user1 }
+  let(:user1) { create :user }
+  let!(:student1) { create :student, name: "John Doe" }
+  let!(:teacher1) { create :teacher, user: user1 }
+  let!(:evaluation1) { create :evaluation, score: Evaluation::RED, teacher: teacher1, student: student1 }
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: Line is too long. [105/100]

Comment (michfarr) [inline spec/features/evaluation_scoring_spec.rb]: Maybe do? :smile:
```ruby
let!(:evaluation1) do
  create(:evaluation, score: Evaluation::Red, teacher: teacher1, student: student1)
end```

Comment (florestankorp) [inline spec/features/evaluation_scoring_spec.rb]: done!

File: app/models/evaluation.rb [hunk 3]
```diff
@@ -26,4 +26,14 @@ def self.pass_params(score, commit)
     @score_params = score
     @commit_params = commit
   end
+
+  def self.score_name
```
Comment (arnoFleming) [inline app/models/evaluation.rb]: Why is this a class method?

Comment (arnoFleming) [inline app/models/evaluation.rb]: Where do you use this method?

File: app/views/teacher/evaluations/show.html.erb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 <div>
   <p><strong>Student: </strong> <%= @evaluation.student.name %></p>
   <p><strong>Score: </strong>
-    <% if @evaluation.score == 0 %>
+    <% if @evaluation.score == Evaluation::RED %>
     <span style="color:red">Red</span>
-    <% elsif @evaluation.score == 1 %>
+    <% elsif @evaluation.score == Evaluation::AMBER %>
```
Comment (arnoFleming) [inline app/views/teacher/evaluations/show.html.erb]: The code I wish you could use, given that `@evaluation` has a method `#color` that responds with `'red'`, `'green'`, or `'amber'`
```html 
<strong>
  Score: 
  <span class="<%= @evaluation.color %>">
    <%= @evaluation.color %>
  </span>
</strong>
```
And then CSS will take care of the rest possibly maybe?

File: spec/features/evaluation_scoring_spec.rb [hunk 5]
```diff
@@ -1,32 +1,39 @@
...
-    let!(:student1) { create :student }
-    let!(:teacher1) { create :teacher, user: user1 }
+  let(:user1) { create :user }
+  let!(:student1) { create :student, name: "John Doe" }
+  let!(:teacher1) { create :teacher, user: user1 }
+  let!(:evaluation1) do
+    create(:evaluation, score: Evaluation::RED, teacher: teacher1, student: student1)
+  end
+
+  context "for user logged in as a teacher" do
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: KISS `context "When logged in as teacher"` (it reveals intent! )

File: spec/features/evaluation_scoring_spec.rb [hunk 6]
```diff
@@ -1,32 +1,39 @@
...
+    it "there is a student named 'John Doe' on the homepage" do
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: Use your exemplars to communicate intent to your fellow programmers

`it "shows all know students"`

File: config/routes.rb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
-  root to: 'evaluation#index'
+  root to: 'teacher/evaluations#index'
```
Comment (arnoFleming) [inline config/routes.rb]: is there a test for this?

File: app/models/evaluation.rb (hunk 3 above)
Comment (arnoFleming) [inline app/models/evaluation.rb]: Is there a test for this method?

File: spec/features/evaluation_scoring_spec.rb [hunk 8]
```diff
@@ -1,32 +1,47 @@
...
-    let(:user1) { create :user }
-    let!(:student1) { create :student }
-    let!(:teacher1) { create :teacher, user: user1 }
+  let(:user1) { create :user }
+  let!(:student1) { create :student, name: "John Doe" }
+  let!(:teacher1) { create :teacher, user: user1 }
+  let!(:evaluation1) do
+    create(:evaluation, score: Evaluation::RED, teacher: teacher1, student: student1)
+  end
+<<<<<<< HEAD
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: unexpected token tLSHFT
(Using Ruby 2.4 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: spec/features/evaluation_scoring_spec.rb [hunk 9]
```diff
@@ -1,32 +1,47 @@
...
+
+  context "When logged in as teacher" do
+    it "redirects to root path" do
+      login_as user1
+      expect(page).to have_current_path(root_path)
+    end
+=======
```
Comment (arnoFleming) [inline spec/features/evaluation_scoring_spec.rb]: unexpected token tEQQ
(Using Ruby 2.4 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: config/routes.rb (hunk 7 above)
Comment (florestankorp) [inline config/routes.rb]: yes

File: app/models/evaluation.rb [hunk 10]
```diff
@@ -27,6 +27,16 @@ def self.pass_params(score, commit)
...
+  def color
+    if score == Evaluation::RED
+      "Red"
+    elsif score == Evaluation::AMBER
+      "Amber"
+    else
+      "Green"
+    end
+  end
+
```
Comment (arnoFleming) [inline app/models/evaluation.rb]: Trailing whitespace detected.

**Pull request merged**